lib/filter.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const urlFn = require('url');
+
+function isExternal(url, config) {
+  let exclude = config.nofollow.exclude;
+  const data = urlFn.parse(url);
+  const host = data.hostname;
+  const sitehost = urlFn.parse(config.url).hostname || config.url;
+
+  if (!data.protocol || !sitehost) return false;
+
+  if (exclude && !Array.isArray(exclude)) exclude = [exclude];
+
+  if (exclude && exclude.length) {
+    for (let i of exclude) {
+      if (sitehost === i) return false;
+    }
+  }
+
+  if (host !== sitehost) return true;
+
+  return false;
+}
+
+module.exports = function(data) {
+  const hexo = this;
+  const config = hexo.config;
+
+  const filterExternal = (data) => {
+    const noFollow = 'external nofollow noreferrer';
+
+    // https://github.com/hexojs/hexo/pull/3685
```

Comment:
I previously added this simply to credit to you, I don't think it's necessary (in this case) since you're now the commit author.